internal/schema/tokmod/token_modifier.go

```diff
@@ -0,0 +1,39 @@
+package tokmod
+
+import (
+	"github.com/hashicorp/hcl-lang/lang"
+)
+
+var (
+	Data              = lang.SemanticTokenModifier("terraform-data")
+	Locals            = lang.SemanticTokenModifier("terraform-locals")
+	Module            = lang.SemanticTokenModifier("terraform-module")
+	Output            = lang.SemanticTokenModifier("terraform-output")
+	Provider          = lang.SemanticTokenModifier("terraform-provider")
+	Resource          = lang.SemanticTokenModifier("terraform-resource")
+	Provisioner       = lang.SemanticTokenModifier("terraform-provisioner")
+	Connection        = lang.SemanticTokenModifier("terraform-connection")
+	Variable          = lang.SemanticTokenModifier("terraform-variable")
+	Terraform         = lang.SemanticTokenModifier("terraform-terraform")
+	Backend           = lang.SemanticTokenModifier("terraform-backend")
+	Name              = lang.SemanticTokenModifier("terraform-name")
+	Type              = lang.SemanticTokenModifier("terraform-type")
+	RequiredProviders = lang.SemanticTokenModifier("terraform-requiredProviders")
+)
```

Comment:
This is probably the most important part of the PR.

It's worth noting that these modifiers - similar to token types - share a global namespace (at least in VSCode) - and so should be prefixed. There doesn't seem to be a way of scoping token modifiers or types to just a single language ID in VSCode.

Two minor changes I'm thinking about here is possibly `terraform-name` => `terraform-label-name` & `terraform-type` => `terraform-label-type`, just to make it more obvious these are used in the context of labels.

Reply:
I would stick with `terraform-type` and `terraform-type` because, as you said, the other choices look kinda unnecessary verbose.